security/s2a/internal/crypter/expander.go

```diff
@@ -0,0 +1,10 @@
+package crypter
+
+import "hash"
+
+// HKDFExpander is the interface for the HKDF expansion function; see
+// https://tools.ietf.org/html/rfc5869 for details. It's use in TLS 1.3 is specified in
+// https://tools.ietf.org/html/rfc8446#section-7.2
+type HKDFExpander interface {
```

Comment:
Why do we need an interface here and can `Expand` be an unexported function in the package where it will be used?

Reply:
I think we'll mostly likely be using it in the `crypter` package only, since I think we're placing everything related to the S2A crypter in this package. I'll unexport both `HKDFExpander` as well as the `Expand`.

As for why we need an interface, @matthewstevenson88 had some ideas about why an interface here may be useful. We can discuss this during standup.

Reply:
We have an interface here in case we need to link in a different crypto library in the future. This way we would need to change any of the logic that depends on HKDF-expand. We can discuss more during the standup?

Reply:
Thanks both for the clarification. Sorry I missed today's standup but let's discuss in tomorrow's.